src/indexable.cr

```diff
@@ -0,0 +1,469 @@
+# A container that allows accessing elements via a numeric index.
```

Comment:
Good idea. I added the docs for this in the module, and in the `unsafe_at` method